scripts/publish.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+# See https://github.com/mapbox/node-cpp-skel
+
+set -eu
+set -o pipefail
+
+export COMMIT_MESSAGE=$(git log --format=%B --no-merges -n 1 | tr -d '\n')
+
+# `is_pr_merge` is designed to detect if a gitsha represents a normal
+# push commit (to any branch) or whether it represents travis attempting
+# to merge between the origin and the upstream branch.
+# For more details see: https://docs.travis-ci.com/user/pull-requests
+function is_pr_merge() {
+  # Get the commit message via git log
+  # This should always be the exactly the text the developer provided
+  local COMMIT_LOG=${COMMIT_MESSAGE}
+
+  # Get the commit message via git show
+  # If the gitsha represents a merge then this will
+  # look something like "Merge e3b1981 into 615d2a3"
+  # Otherwise it will be the same as the "git log" output
+  export COMMIT_SHOW=$(git show -s --format=%B | tr -d '\n')
+
+  if [[ "${COMMIT_LOG}" != "${COMMIT_SHOW}" ]]; then
+     echo true
+  fi
+}
+
+# `publish` is used to publish binaries to s3 via commit messages if:
+# - the commit message includes [publish binary]
+# - the commit message includes [republish binary]
+# - the commit is not a pr_merge (checked with `is_pr_merge` function)
+function publish() {
+  echo "dumping binary meta..."
+  ./node_modules/.bin/node-pre-gyp reveal --loglevel=error $@
+
+  echo "determining publishing status..."
+
+  if [[ $(is_pr_merge) ]]; then
+      echo "Skipping publishing because this is a PR merge commit"
+  else
+      echo "Commit message: ${COMMIT_MESSAGE}"
+
+      if [[ ${COMMIT_MESSAGE} =~ "[publish binary]" ]]; then
+          echo "Publishing"
+          ./node_modules/.bin/node-pre-gyp package publish $@
+      elif [[ ${COMMIT_MESSAGE} =~ "[republish binary]" ]]; then
```

Comment:
I think I read somewhere that republish does not work on NPM, since they do not want to overwrite existing binaries. Did you check whether republishing is possible? (https://docs.npmjs.com/cli/unpublish)